example/src/main/java/com/stripe/example/activity/SimplePaymentMethodConfirmationActivity.kt

```diff
@@ -0,0 +1,308 @@
+        )[SimplePaymentMethodConfirmationViewModel::class.java]
+    }
+
+    private val dropdownItem: DropdownItem
+        get() {
+            return DropdownItem.valueOf(
+                viewBinding.paymentMethod.text.toString()
+            )
+        }
+
+    private val paymentMethodCreateParams: PaymentMethodCreateParams
+        get() {
+            val dropdownItem = this.dropdownItem
+            val billingDetails = PaymentMethod.BillingDetails(
+                name = viewBinding.name.text.toString().takeIf { dropdownItem.requiresName },
+                email = viewBinding.email.text.toString().takeIf { dropdownItem.requiresEmail }
+            )
+            return dropdownItem.createParams(billingDetails, null)
+        }
+
+    private fun onDropdownItemSelected() {
+        val dropdownItem = this.dropdownItem
+        viewBinding.nameLayout.visibility = viewVisibility(dropdownItem.requiresName)
+        viewBinding.emailLayout.visibility = viewVisibility(dropdownItem.requiresEmail)
+    }
+
+    private fun viewVisibility(visible: Boolean): Int {
+        return if (visible) {
+            View.VISIBLE
+        } else {
+            View.GONE
+        }
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(viewBinding.root)
+
+        viewModel.inProgress.observe(this, Observer { enableUi(!it) })
+        viewModel.status.observe(this, Observer(viewBinding.status::setText))
+
+        val adapter = DropdownItemAdapter(this)
+        viewBinding.paymentMethod.setAdapter(adapter)
+        viewBinding.paymentMethod.setOnItemClickListener { _, _, _, _ ->
+            viewModel.status.value = ""
+            onDropdownItemSelected()
+        }
+        viewBinding.paymentMethod.setText(DropdownItem.P24.name, false)
+        onDropdownItemSelected()
+
+        viewBinding.payNow.setOnClickListener {
+            keyboardController.hide()
+
+            viewModel.createPaymentIntent(dropdownItem.country) {
+                handleCreatePaymentIntentResponse(it, paymentMethodCreateParams)
+            }
+        }
+    }
+
+    override fun onSaveInstanceState(outState: Bundle) {
+        super.onSaveInstanceState(outState)
+        outState.putString(STATE_EMAIL, viewBinding.email.text.toString())
+        outState.putString(STATE_NAME, viewBinding.name.text.toString())
+        outState.putString(STATE_DROPDOWN_ITEM, viewBinding.paymentMethod.text.toString())
+    }
+
+    override fun onRestoreInstanceState(savedInstanceState: Bundle) {
+        super.onRestoreInstanceState(savedInstanceState)
+        savedInstanceState.getString(STATE_EMAIL)?.let(viewBinding.email::setText)
+        savedInstanceState.getString(STATE_NAME)?.let(viewBinding.name::setText)
+        savedInstanceState.getString(STATE_DROPDOWN_ITEM)?.let {
+            viewBinding.paymentMethod.setText(it)
+            onDropdownItemSelected()
+        }
+    }
```

Comment:
this is totally fine, but you could also take advantage of the `ViewModel` to preserve state across rotation

Reply:
Yeah, I looked into doing that but it didn't really seem any better than just doing this because the data isn't coming from the view model (it's coming from the user). I'd have to add a text changed listener to each view and store it on the view model every time something changes. Or am I missing something here?

Reply:
You could add the vars on the ViewModel and populate when `onSaveInstanceState()` is called, then read and update view in `onCreate()`. There's probably no good reason to do this here though.